planner/core/logical_plans.go

```diff
@@ -536,8 +581,19 @@ func (ds *DataSource) Convert2Gathers() (gathers []LogicalPlan) {
 			// If index columns can cover all of the needed columns, we can use a IndexGather + IndexScan.
 			if isCoveringIndex(ds.schema.Columns, path.FullIdxCols, path.FullIdxColLens, ds.tableInfo.PKIsHandle) {
 				gathers = append(gathers, ds.buildIndexGather(path))
+			} else {
+				needBuildDoubleGather := true
+				for _, idxCol := range path.FullIdxCols {
+					if idxCol == nil {
+						needBuildDoubleGather = false
```

Comment:
Why?

Reply:
I'll check this one, maybe we don't need the `needBuildDoubleGather`.

Reply:
Done